pkg/piecestore/rpc/client/client.go

```diff
@@ -16,10 +16,11 @@ import (
 	pb "storj.io/storj/protos/piecestore"
 )
 
+// PSClient is an interface describing the functions for interacting with piecestore nodes
 type PSClient interface {
 	Meta(ctx context.Context, id PieceID) (*pb.PieceSummary, error)
-	Put(ctx context.Context, id PieceID, data io.Reader, ttl time.Time) error
-	Get(ctx context.Context, id PieceID, size int64) (ranger.RangeCloser, error)
+	Put(ctx context.Context, id PieceID, size int64, data io.Reader, ttl time.Time, payer, clientBandWidthAlloc string) error
```

Comment:
we don't know the size upfront though. we may have a promise-to-pay signed by the payer for up to a certain amount though.

imagine these protobufs:
```
client sends: okay bridge says i can send up to 64MB.
client sends: okay after the next packet I will have sent 1KB. okay here's 1KB
client sends: okay after the next packet I will have sent 2KB. okay here's 1KB
client sends: okay after the next packet I will have sent 3KB. okay here's 1KB
client sends: okay after the next packet I will have sent 4KB. okay here's 1KB
client sends: okay after the next packet I will have sent 5KB. okay here's 1KB
```
by the end, the farmer has a document saying the client used 5KB, even though no one knew up front.

reasoning: bridge wants to only be talked to up front once per segment about what it's on the hook to pay.
client is already sending multiple packets. in each packet just send an update.

if the client sends a huge thing like "okay after the next packet i will have downloaded 64MB", a malicious farmer can save that promise-to-pay and then shut the connection down without receiving that data, sending the promise-to-pay up to the bridge. so we want the individual promises-to-pay to be small so that problem is limited, even if the requested file is large